routers/api/packages/generic/generic.go

```diff
@@ -97,7 +95,6 @@ func UploadPackage(ctx *context.Context) {
 				Name:        packageName,
 				Version:     packageVersion,
 			},
-			SemverCompatible: true,
```

Comment:
Can version still be parsed and if there is no error set SemverCompatible as true?

Reply:
👍

Reply:
What would the benefit of doing so be? Does Gitea do anything special if the version is SemVer?

What happens if one version of the same package happens to be SemVer-compatible by accident, but other versions of it aren't?

Reply:
Why is that property there if it's not used? Imho it could be useful when comparing versions probably but I don't know

Reply:
NuGet packages for example use SemVer and it's used to sort the versions when creating the package list responses.

Reply:
> Why is that property there if it's not used? Imho it could be useful when comparing versions probably to compare only versions that are semver compatible

It's not there in the latest push, the line we're commenting on *removes* it.

I originally changed it to false, but there was a suggestion here to just remove it, since false is the default, so with the suggestion applied, this line in particular is gone.

> NuGet packages for example use SemVer and it's used to sort the versions when creating the package list responses.

Yeah, but this is the Generic repo, not NuGet. NuGet enforces SemVer, so sorting by SemVer makes sense. Sorting by SemVer when it's opportunistic at best would not make sense. How do you sort "1.0.0" (semver-compatible), "1.0.0q" (non semver)? Semver would choke on that, so we'd have to fall back on plain sorting.

If SemVer enforcement is to be an option for the generic repo, then it should be an owner-wide, or gitea-wide option, rather than opportunistic, to avoid surprises.

Reply:
> Yeah, but this is the Generic repo, not NuGet. NuGet enforces SemVer, so sorting by SemVer makes sense. Sorting by SemVer when it's opportunistic at best would not make sense. How do you sort "1.0.0" (semver-compatible), "1.0.0q" (non semver)? Semver would choke on that, so we'd have to fall back on plain sorting.
> 
> If SemVer enforcement is to be an option for the generic repo, then it should be an owner-wide, or gitea-wide option, rather than opportunistic, to avoid surprises.

I was just answering lafriks answer. There is no reason to enforce SemVer for generic packages.